numpy/ma/core.pyi

```diff
@@ -465,8 +466,78 @@ class MaskedArray(ndarray[_ShapeType_co, _DType_co]):
     def std(self, axis=..., dtype=..., out=..., ddof=..., keepdims=...): ...
     def round(self, decimals=..., out=...): ...
     def argsort(self, axis=..., kind=..., order=..., endwith=..., fill_value=..., *, stable=...): ...
-    def argmin(self, axis=..., fill_value=..., out=..., *, keepdims=...): ...
-    def argmax(self, axis=..., fill_value=..., out=..., *, keepdims=...): ...
+    @overload
+    def argmin(  # type: ignore[override]
+        self,
+        axis: None = None,
+        fill_value: _ScalarLike_co | None = None,
+        out: None = None,
+        *,
+        keepdims: Literal[False] | _NoValueType = ...,
+    ) -> intp: ...
+    @overload
+    def argmin(  # type: ignore[override]
+        self,
+        axis: _ShapeLike | None = None,
+        fill_value: _ScalarLike_co | None = None,
+        out: None = None,
+        *,
+        keepdims: bool | _NoValueType = ...,
+    ) -> Any: ...
+    @overload
+    def argmin(  # type: ignore[override]
+        self,
+        axis: _ShapeLike | None = None,
+        fill_value: _ScalarLike_co | None = None,
+        *,
+        out: _ArrayType,
+        keepdims: bool | _NoValueType = ...,
+    ) -> _ArrayType: ...
```

Comment:
After experimenting a bit with this, I noticed that safe casting rules apply to the `out` dtype. In practice that means that the scalar-type of `out` can only be `np.bool` or `np.integer`. You could consider using this fact to narrow the type that `out` accepts. It's already technically correct right now, so I consider it an optional improvement.

Reply:
yup, thanks! this is also an issue for `np.argmin`, so i've opened https://github.com/numpy/numpy/issues/28641 as a follow-up if that's ok